app/src/main/kotlin/com/wire/android/ui/authentication/create/code/CodeScreen.kt

```diff
@@ -0,0 +1,113 @@
+package com.wire.android.ui.authentication.create.personalaccount
+
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.interaction.MutableInteractionSource
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.fillMaxHeight
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.material3.ExperimentalMaterial3Api
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Scaffold
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.input.TextFieldValue
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.text.style.TextDecoration
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import com.wire.android.R
+import com.wire.android.ui.authentication.create.code.CodeViewModel
+import com.wire.android.ui.authentication.create.code.CodeViewState
+import com.wire.android.ui.common.textfield.CodeTextField
+import com.wire.android.ui.common.textfield.WireTextFieldState
+import com.wire.android.ui.common.topappbar.WireCenterAlignedTopAppBar
+import com.wire.android.ui.theme.wireDimensions
+import com.wire.android.ui.theme.wireTypography
+
+@Composable
+fun CodeScreen(viewModel: CodeViewModel, title: String) {
+    CodeContent(
+        state = viewModel.codeState,
+        title = title,
+        onCodeChange = viewModel::onCodeChange,
+        onResendCodePressed = viewModel::onResendCodePressed,
```

Comment:
small detail I think it is a personal test, but a viewmodel function should not indicate the view action, I do not think it matters for the Viewmodel if the button is pressed or long hold or a switch or a toggle is used, resendCode will be enough I think

Reply:
Sure, right 🚀